controller/Diagnostic.php

```diff
@@ -0,0 +1,211 @@
+     * It also allows launching new ones.
+     */
+    public function index()
+    {
+        $diagnostics       = $this->getDiagnosticDataTable()->getDiagnostics($this->getRequestOptions());
+
+        $data = array(
+            'title'              => __('Readiness diagnostics'),
+            'set'                => $diagnostics,
+            'config'             => $this->loadConfig(),
+        );
+
+        foreach($data as $key => $value) {
+            if (is_array($value) || is_object($value)) {
+                $data[$key] = json_encode($value);
+            }
+        }
+
+        $userLabel = SessionManager::getSession()->getUserLabel();
+
+        $this->setData('cls', 'diagnostic-index');
+        $this->setData('clientConfigUrl', $this->getClientConfigUrl());
+        $this->setData('userLabel', $userLabel);
+
+        $this->defaultData();
+        $this->setData('data', $data);
+        $this->setData('content-template', 'pages/index.tpl');
+        $this->setView('layout.tpl');
+    }
+
+    /**
+     * Display the diagnostic runner
+     */
+    public function diagnostic()
+    {
+        $data = array(
+            'title'  => __('Readiness Check'),
+            'config' => $this->loadConfig(),
+        );
+
+        foreach($data as $key => $value) {
+            if (is_array($value) || is_object($value)) {
+                $data[$key] = json_encode($value);
+            }
+        }
+
+        $this->defaultData();
+        $this->setData('userLabel', SessionManager::getSession()->getUserLabel());
+        $this->setData('clientConfigUrl', $this->getClientConfigUrl());
+        $this->setData('cls', 'diagnostic-runner');
+        $this->setData('data', $data);
+        $this->setData('content-template', 'pages/index.tpl');
+        $this->setView('layout.tpl');
+    }
+
+    /**
+     * Gets the list of diagnostic results
+     *
+     * @throws \common_Exception
+     */
+    public function diagnosticData()
+    {
+        try {
+            $this->returnJson($this->getDiagnosticDataTable()->getDiagnostics($this->getRequestOptions()));
+        } catch (ServiceNotFoundException $e) {
+            \common_Logger::w(__('No diagnostic service defined'));
+            $this->returnError(__('Interface not available'));
+        }
+    }
+
+    /**
+     * Removes diagnostic results
+     *
+     * @throws \common_Exception
+     */
+    public function remove()
+    {
+        $id = $this->getRequestParameter('id');
+        $this->returnJson([
+            'success' => $this->getDiagnosticDataTable()->removeDiagnostic($id)
+        ]);
+    }
+
+    /**
+     * Gets the data table request options
+     *
+     * @param array $defaults
+     * @return array
+     */
+    protected function getRequestOptions(array $defaults = []) {
+
+        $defaults = array_merge($this->getDefaultOptions(), $defaults);
+
+        $page = $this->hasRequestParameter('page') ? $this->getRequestParameter('page') : $defaults['page'];
```

Comment:
Already discussed with JS, he will refactor this part to improve that and able to work with paginated storage